index.js

```diff
@@ -1,14 +1,128 @@
-const express = require('express');
-const helmet = require('helmet');
+require("dotenv").config();
+
+const express = require("express");
+const helmet = require("helmet");
+const morgan = require("morgan");
+const knex = require("knex");
+
+const knexConfig = {
+  client: "sqlite3",
+  connection: {
+    filename: "./data/lambda.sqlite3"
+  },
+  useNullAsDefault: true
+};
+const db = knex(knexConfig);
 
 const server = express();
 
 server.use(express.json());
 server.use(helmet());
+server.use(morgan("dev"));
 
 // endpoints here
 
-const port = 3300;
-server.listen(port, function() {
+// Get all zoos
+server.get("/api/zoos", async (req, res) => {
+  try {
+    const zoos = await db("zoos");
+    res.status(200).json(zoos);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({
+      error,
+      message: "Failed to retrieve zoo data"
+    });
+  }
+});
+
+// Get zoo by id
+server.get("/api/zoos/:id", async (req, res) => {
+  try {
+    const zoo = await db("zoos")
+      .where({ id: req.params.id })
```

Comment:
We need some logic somewhere here that will deliver a 404 error message if we cannot find the information at the id specified. Do you know how we could achieve this?